sets.h

```diff
@@ -0,0 +1,46 @@
+/*
+    Create sets.
+
+    This file is part of libscanmem.
+
+    This library is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published
+    by the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with this library.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef SETS_H
+#define SETS_H
+
+#include <stdlib.h>
+
+struct set {
+    size_t sz;  /* size of set (used) */
+    void  *buf; /* value buffer       */
+};
+
+inline void set_free(struct set *set)
+{
+    if (set)
+        free(set->buf);
+    free(set);
+}
+
+/* iterate over set from front (forwards) */
+#define foreach_set_fw(i, set) \
+    for (size_t i = 0; i < (set)->sz; i++)
```

Comment:
This is C11 only, you can't define vars in the `for` scope in C99.

Either make another subscope or, possibly for the better, drop the macros and leave looping to the caller, the interface is simple enough that hiding it does not really help much.

Reply:
Vars in the for loop are a C99 extension. I think it is probably a good idea to keep the macros for the unsigned sets, as iterating backwards can be tricky.

Reply:
Shame on me, I was 100% sure they were C11.
Ignore this whole comment block then.